src/plugins/intel_npu/src/plugin/npuw/util.cpp

```diff
@@ -234,6 +234,18 @@ inline float avx2_load_f32(const int8_t* data, ov::element::Type type) {
     }
 }
 
+inline __m256 u4_avx2_load_f32(const uint8_t* data) {
+    float unpacked[8];
+    for (int i = 0; i < 8; ++i) {
+        uint8_t u4_element = lo4(data[i]);
```

Comment:
Also here you take `lo` but never take `hi`. so either you read 8 bytes and produce 16 elements or you need to read 4 bytes to produce 8.